### ISSUE TITLE
Detect "who" asked for something to Alexa

### ISSUE THREAD

**Issue by jesusmarcos** [custom-components/alexa_media_player#455]
I find the sensor.last_alexa very useful for creating routines like "Switch on the light" and then (from homeassistant) detect which Alexa was the last, and switch on the proper light.

It could be nice to have another sensor (in case it's possible) that detects which person executed a voice command (Alexa can recognize different voices assigned to Amazon's account). I would like to have it to create routines like "Switch on the light of my room", so it detects who am I and then control it from homeassistant.

Is this information available somewhere so you can create a sensor with it?

Thanks a lot in advance

Jesús

**Comment by alandtse** [comment]
I don't have this enabled so I don't know if the data is available.  Can you see any data from this [page (fix domain region)](https://alexa.amazon.com/api/activities?startTime=&size=10&offset=1) that shows voice info?

**Comment by jesusmarcos** [comment]
Ok, thanks. I was not aware of this page.

I got information from there from different devices and different "voices" but I couldn't find any match regarding to the voice. I could find, the device, the customer_id,... but not the person that launched the command.

Bad luck. :-(

Thanks for your time. Alexa Media it's a very valuable component for me.

I close the request

**Comment by alandtse** [comment]
Maybe it's in some other field like the `id`? We know the `id` contains some fields separated by a `#` and it's possible the voice somewhere in there? For example, using the devices [page (fix region)](https://alexa.amazon.com/api/devices-v2/device) I can identify the following fields:

```
[deviceOwnerCustomerId]#[unknown but maybe just the activity id]#[deviceType]#[serialNumber]
```

I wonder if the `deviceOwnerCustomerId` actually refers to the identified voice account and I've misidentified it because it defaults to the owner when no voice id is enabled?

Assuming it was, we'd then have to figure out how to identify accounts. This [page (fix region)](https://alexa.amazon.com/api/household) may help.

As a note, the [wiki](https://github.com/custom-components/alexa_media_player/wiki/Known-Endpoints) lists pages we know about in case you wanted to try exploring.

**Comment by jesusmarcos** [comment]
Yes, I also check it. I have 3 examples (obtained from your first page):

- Device 1 and my voice.
- Device 2 and my voice.
- Device 1 and my wife's voice.

We trained Alexa so it's able to recognise us (sometimes respond with our names) but all the Alexas are created in my account.

Can I send you privately the logs?  (I wouldn't like to publish this information but I can send it to you in case it helps). I think I verified all the fields and each part of them but maybe you will see it differently.

Thanks a lot for your interest

Jesus
